service/controller/clusterapi/v19/resources/app/desired.go

```diff
@@ -0,0 +1,96 @@
+package app
+
+import (
+	"context"
+	"strconv"
+
+	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cmav1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/giantswarm/cluster-operator/pkg/annotation"
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	"github.com/giantswarm/cluster-operator/pkg/project"
+	pkgkey "github.com/giantswarm/cluster-operator/pkg/v19/key"
+	awskey "github.com/giantswarm/cluster-operator/service/controller/aws/v19/key"
+	azurekey "github.com/giantswarm/cluster-operator/service/controller/azure/v19/key"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v19/key"
+	kvmkey "github.com/giantswarm/cluster-operator/service/controller/kvm/v19/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8sv1alpha1.App, error) {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var apps []*g8sv1alpha1.App
+
+	for _, appSpec := range r.newAppSpecs() {
+		apps = append(apps, r.newApp(cr, appSpec))
+	}
+
+	return apps, nil
+}
+
+func (r *Resource) newApp(cr cmav1alpha1.Cluster, appSpec pkgkey.AppSpec) *g8sv1alpha1.App {
+	return &g8sv1alpha1.App{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ChartConfig",
+			APIVersion: "application.giantswarm.io",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				annotation.ForceHelmUpgrade: strconv.FormatBool(appSpec.UseUpgradeForce),
+			},
+			Labels: map[string]string{
+				label.App:                appSpec.App,
+				label.AppOperatorVersion: "1.0.0",
```

Comment:
We had this for the `cert-operator` version as well. How can we make this better in the future? We are still hiding release related versions hardcoded hidden in the depths of the projects. Same for `node-operator` in the provider operators. I did this for the [`certconfig` resource](https://github.com/giantswarm/cluster-operator/blob/6c79fabae25bcf5f4fa210b666aa115aad191594/service/controller/clusterapi/v19/resources/certconfig/desired.go#L69). 

```
Version: cc.Status.Versions["cert-operator.giantswarm.io/version"],
```

There is a resource fetching versions and puts them into the controller context. Can we do this here as well? Then it is at least wired dynamically and part of a release.

Reply:
OK cool I'll check on that. As yes having this wired dynamically would be better.

Reply:
@xh3b4sd OK I'm getting there but I still have questions!

https://github.com/giantswarm/cluster-operator/blob/6c79fabae25bcf5f4fa210b666aa115aad191594/service/controller/clusterapi/v19/resources/operatorversions/resource.go#L61-L73

Where are the version bundles in the release defined? Also do I need to add app-operator or is it already there?

Reply:
Good question. Is it that we need to add the operator in installations so that it is just in the yaml files and then it is automatically parsed? /cc @tuommaki

Reply:
Definition of what version bundles belong to release is currently in `installations` (see authorities: https://github.com/giantswarm/installations/blob/master/release/provider/aws.yaml#L3). When your operator is deployed and provides the versionbundle endpoint, then it can be listed there and it will be part of response in above code.

Reply:
Nice thanks. I'll add app-operator there. 👍